AWS: Updated AWS authentication documentation. (#39236)
grafana/grafana: docs/sources/datasources/aws-cloudwatch/aws-authentication.md

weight = 205
+++

> ## AWS Authentication

Reviewer: ```suggestion
## AWS authentication
```

> ## AWS authentication

Reviewer 2: ```suggestion
# AWS authentication
```

> Requests from a Grafana plugin to AWS is made on behalf of an [IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html) or an [IAM user](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users.html). The IAM user or IAM role needs to have the permission policies needed in order to perform a certain api action. The policies needed are specific to each data source, so please refer to the specific data source documentation for details.

Reviewer 2: ```suggestion
Requests from a Grafana plugin to AWS are made by an IAM role or an IAM user. The IAM user or IAM role must have the associated permission and associated policies to perform certain API actions. Since these policies are specific to each data source, refer to the data source documentation for details.
```

Author: Looks good, but I think we should use the words `on behalf of` since that's how they usually describe this in the AWS docs.

All requests to AWS api:s are done server side by the Grafana backend using the official AWS SDK.

> # Authentication methods

Reviewer 2: ```suggestion
## Authentication methods
```

- `Credentials file` corresponds directly to the [SharedCredentialsProvider](https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/#SharedCredentialsProvider) provider in the Go SDK. It reads the AWS shared credentials file to find a given profile. While `AWS SDK Default` will also find the shared credentials file, this option allows you to specify which profile to use without using environment variables. This option doesn't have any implicit fallbacks to other credential providers, and it fails if the credentials provided from the file aren't correct.

- `Access and secret key` corresponds to the [StaticProvider](https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/#StaticProvider) and uses the given access key ID and secret key to authenticate. This method doesn't have any fallbacks, and will fail if the provided key pair doesn't work.

> - `Workspace IAM role` corresponds to the [EC2RoleProvider](https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/ec2rolecreds/#EC2RoleProvider). The `EC2RoleProvider` attempts to pull credentials for the role that is attached to the EC2 instance that Grafana runs on. This can also be achieved by using the authentication method `AWS SDK Default`, but this option is different in the sense that it doesn't have any fallbacks. This option is currently only enabled by default in Amazon Managed Grafana.

Reviewer 2: ```suggestion
- `Workspace IAM role` corresponds to the [EC2RoleProvider](https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/ec2rolecreds/#EC2RoleProvider). The EC2RoleProvider pulls credentials for a role attached to the EC2 instance that Grafana runs on. You can also achieve this by using the authentication method AWS SDK Default, but this option is different as it doesn’t have any fallbacks. This option is currently only enabled by default in Amazon Managed Grafana.
```

## Assuming a role

The `Assume Role ARN` field allows you to specify which IAM role to assume. When left blank, the provided credentials are used directly and the associated role or user should have the required permissions. If this field is non-blank, on the other hand, the provided credentials are used to perform an [sts:AssumeRole](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html) call.

> It's possible to disable this feature in the Grafana configuration. For more information, refer to [assume_role_enabled]({{< relref "../../administration/configuration.md#assume_role_enabled" >}}) documentation.

Reviewer 2: ```suggestion
You can disable this feature in the Grafana configuration. For more information, refer to [assume_role_enabled]({{< relref "../../administration/configuration.md#assume_role_enabled" >}}) documentation.
```

### External ID

> If you are assuming a role in another account that has been created with an external ID, specify the external ID in this field. For more information, please refer to the [AWS documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html).

Reviewer 2: ```suggestion
If you are assuming a role in another account that was created with an external ID, then specify the external ID in this field. For more information, refer to the [AWS documentation on external ID](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html).
```

> ## Endpoint

Reviewer 2: Should this be H3?

Author: Yeah think it should since this is an addition to the assume role feature.

# AWS authentication

Requests from a Grafana plugin to AWS is made on behalf of an [IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html) or an [IAM user](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users.html). The IAM user or IAM role needs to have the permission policies needed in order to perform a certain api action. The policies needed are specific to each data source, so please refer to the specific data source documentation for details.

> All requests to AWS api:s are done server side by the Grafana backend using the official AWS SDK.

Reviewer 2: ```suggestion
All requests to AWS APIs are performed on the server side by the Grafana backend using the official AWS SDK.
```
